pyro/distributions/zero_inflated.py

```diff
@@ -110,3 +114,15 @@ def __init__(self, gate, total_count, probs=None, logits=None, validate_args=Non
         super().__init__(
             gate, base_dist, validate_args=validate_args
```

Comment:
Now that `arg_constraints` is specified, args will be validated twice. I think we could avoid double validation similar to how it's done in `.expand()`, i.e.
```py
super().__init__(gate, base_dist, validate_args=False)
self._validate_args = validate_args
```
WDYT?

Reply:
I guess you meant `base_dist = NegativeBinomial(..., validate_args=False)`? We need to validate the `gate` too.

Reply:
Hmm ok that version works too, just make sure you set `._validate_args` afterwards so that `.log_prob()` will validate:
```py
base_dist = NegativeBinomial(..., validate_args=False)
base_dist._validate_args = validate_args
```

Reply:
You are right, thanks!